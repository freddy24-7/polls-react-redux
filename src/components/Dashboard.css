.center {
    text-align: center;
    padding: 10px;
}

.h2 {
    text-align: center;
    padding: 10px;
    margin-top: 30px;
}

.h5 {
    text-align: center;
    padding: 10px;
    margin-top: 20px;
}

.dashboard-container {
    background-color: #f1f1f1;
    min-height: 100vh;
    text-align: center;
}

.toggle-container {
    margin-top: 20px;
    text-align: center;
}

.text {
    text-align: center;
    color: #ba8b1e
}

.toggle-container {
    margin-top: 20px;
    text-align: center;
    color: #ba8b1e

}
