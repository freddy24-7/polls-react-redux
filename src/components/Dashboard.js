import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import useInitialDataLoader from '../utils/initialDataLoader';
import Toggle from 'react-toggle';
import 'react-toggle/style.css';
import './Dashboard.css';
import QuestionList from './QuestionList';

const Dashboard = () => {
  const userId = useSelector((state) => state.authedUser);
  const questions = useSelector((state) => state.questions);
  const users = useSelector((state) => state.users);
  const user = users[userId];
  const [showAnswered, setShowAnswered] = useState(true); // State to toggle between answered and unanswered questions

  // Custom hook to load initial data
  useInitialDataLoader();

  // Computing answered and unanswered question IDs
  const answeredQuestionIds = user ? Object.keys(user.answers) : [];
  const unansweredQuestionIds = Object.keys(questions).filter(
    (id) => id !== 'newQuestion' && !answeredQuestionIds.includes(id),
  );

  return (
    <div className="dashboard-container">
      <h5 className="h5">Welcome {user ? user.name : ''}</h5>
      <div className="toggle-container">
        <label>
          <Toggle
            checked={showAnswered}
            onChange={() => setShowAnswered(!showAnswered)}
          />
          <span>
            <p className="text">Show Responded To / Show New Questions</p>
          </span>
        </label>
      </div>
      {showAnswered && answeredQuestionIds.length > 0 && (
        <h2 className="h2">Questions Responded To</h2>
      )}
      {showAnswered && (
        <QuestionList
          questionIds={answeredQuestionIds}
          answeredQuestionIds={answeredQuestionIds}
        />
      )}
      {!showAnswered && unansweredQuestionIds.length > 0 && (
        <h2 className="center">New Questions</h2>
      )}
      {!showAnswered && (
        <QuestionList
          questionIds={unansweredQuestionIds}
          answeredQuestionIds={answeredQuestionIds}
        />
      )}
    </div>
  );
};

export default Dashboard;
