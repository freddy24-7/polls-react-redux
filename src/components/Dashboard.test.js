import React from 'react';
import '@testing-library/jest-dom';
import Dashboard from './Dashboard';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);
const store = mockStore({
    authedUser: 'user123', // Provide any initial state values needed for testing
    questions: {},
    users: {}
});

test('renders correctly', () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <Dashboard />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});