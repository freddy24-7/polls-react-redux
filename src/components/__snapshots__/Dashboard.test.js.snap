// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dashboard-container"
>
  <h5
    className="h5"
  >
    Welcome 
  </h5>
  <div
    className="toggle-container"
  >
    <label>
      <div
        className="react-toggle react-toggle--checked"
        onClick={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="react-toggle-track"
        >
          <div
            className="react-toggle-track-check"
          >
            <svg
              height="11"
              viewBox="0 0 14 11"
              width="14"
            >
              <path
                d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                fill="#fff"
                fillRule="evenodd"
              />
            </svg>
          </div>
          <div
            className="react-toggle-track-x"
          >
            <svg
              height="10"
              viewBox="0 0 10 10"
              width="10"
            >
              <path
                d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                fill="#fff"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </div>
        <div
          className="react-toggle-thumb"
        />
        <input
          checked={true}
          className="react-toggle-screenreader-only"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
      </div>
      <span>
        <p
          className="text"
        >
          Show Responded To / Show New Questions
        </p>
      </span>
    </label>
  </div>
</div>
`;
